ompi/mpi/fortran/use-mpi-f08/mod/mpi-f08-types.F90

```diff
@@ -78,47 +79,47 @@ module mpi_f08_types
   ! Pre-defined handles
   !
 
-  type(MPI_Comm),       bind(C, name="ompi_f08_mpi_comm_world") OMPI_PROTECTED      :: MPI_COMM_WORLD
-  type(MPI_Comm),       bind(C, name="ompi_f08_mpi_comm_self") OMPI_PROTECTED        :: MPI_COMM_SELF
+  type(MPI_Comm), parameter       :: MPI_COMM_WORLD              = MPI_Comm(OMPI_MPI_COMM_WORLD)
+  type(MPI_Comm), parameter       :: MPI_COMM_SELF               = MPI_Comm(OMPI_MPI_COMM_SELF)
```

Comment:
So is the overall idea of this PR to take what used to be instances declared in C and turn them into pure Fortran variables?

E.g., MPI_COMM_SELF becomes a parameter (which, IIRC, effectively means constant, right?), and initially define it to be equal to a new `MPI_Comm()` object, essentially initialized with `OMPI_MPI_COMM_SELF`?

If so, is it correct to assume that this is standard F08 notation, and is not specific to any particular compiler?